tests/client.rs

```diff
@@ -248,3 +250,43 @@ fn test_accept_header_is_not_changed_if_set() {
 
     assert_eq!(res.status(), &reqwest::StatusCode::Ok);
 }
+
+#[test]
+fn test_gzip_response() {
+    let mut e = ::flate2::write::GzEncoder::new(Vec::new(), flate2::Compression::Default);
+    e.write(b"test request");
+
+    let gzipped_content = e.finish().unwrap();
+    let mut response_d = b"\
+            HTTP/1.1 200 OK\r\n\
+            Server: test-accept\r\n\
+            Content-Encoding: gzip\r\n\
+            Content-Length: 0\r\n\r\n".to_vec();
+
+    response_d.extend(&gzipped_content);
+    let server = server! {
+        request: b"\
+            GET /gzip HTTP/1.1\r\n\
+            Host: $HOST\r\n\
+            User-Agent: $USERAGENT\r\n\
+            Accept: */*\r\n\
+            \r\n\
+            ",
+        response: b"\
+            HTTP/1.1 200 OK\r\n\
+            Server: test-accept\r\n\
+            Content-Encoding: gzip\r\n\
+            Content-Length: 0\r\n\
```

Comment:
You're telling reqwest that the length of the response body is 0, and hyper will enforce this. After reading the headers, when trying to read the body from hyper, it will report that the body is empty.

- So, the easy fix is to change this to the length of the body (in this case, 105) (a wonderful example of how compression actually explodes small values, 12 -> 105!).
- I'm wondering if this specific case should also be fix. If there is no body, should the read just return `Ok(0)`? Or should point out that decoding an empty body doesn't work, and so the response is malformed? I'm probably leaning towards don't error (however, a `warn!` log may be appropriate.)